### PR TITLE
add fedora support

### PR DESCRIPTION
**Pull request by vjayaramrh**
This PR adds support for Fedora34 that has the kernel version of 5.17.6. Was able to execute the steps at https://github.com/nephio-project/test-infra/blob/main/e2e/provision/README.md successfully.

Note:
-  Fedora versions above 34 have kernel version greater than 6.0.0 which does not support building and installing gtp5g kernel modules.
-  PR was tested on a Fedora VM running on baremetal
- Fedora 34 VM on gcp cloud does not allow ssh login and hence was unable to test out in gcp fedora34 VM.

### CODE REVIEW

File: e2e/provision/playbooks/roles/bootstrap/vars/redhat.yml
Comment (gvbalaji) [inline e2e/provision/playbooks/roles/bootstrap/vars/redhat.yml:1]: What's this file for?  If this is needed can we rename the file to a  more neutral name?

**Review by gvbalaji — APPROVED** [review]
Thanks @vjayaramrh . Added a comment but overall looks good.

File: e2e/provision/install_sandbox.sh [hunk 1]
```diff
@@ -29,9 +29,17 @@ function get_status {
...
+        sudo apt-get update
+        sudo -E DEBIAN_FRONTEND=noninteractive apt-get remove -q -y python3-openssl
+        sudo -E NEEDRESTART_SUSPEND=1 DEBIAN_FRONTEND=noninteractive apt-get -o Dpkg::Options::="--force-confnew" --allow-downgrades --allow-remove-essential --allow-change-held-packages -fuy install -q -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" python3-pip
+elif [ -f /etc/redhat-release ]; then
+        sudo dnf install python3-pip -y
+else
+        echo "OS not supported"
+        exit
+fi
+
```
Comment (electrocucaracha) [inline e2e/provision/install_sandbox.sh:42]: Same thing here
```suggestion
    # shellcheck disable=SC1091
    source /etc/os-release || source /usr/lib/os-release
    case ${ID,,} in
    ubuntu | debian)
        sudo apt-get update
        sudo -E DEBIAN_FRONTEND=noninteractive apt-get remove -q -y python3-openssl
        sudo -E NEEDRESTART_SUSPEND=1 DEBIAN_FRONTEND=noninteractive apt-get -o Dpkg::Options::="--force-confnew" --allow-downgrades --allow-remove-essential --allow-change-held-packages -fuy install -q -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" python3-pip
        ;;
    rhel | centos | fedora | rocky)
        PKG_MANAGER=$(command -v dnf || command -v yum)
        sudo $PKG_MANAGER install python3-pip -y
        ;;
    *)
        echo "OS not supported"
        exit
        ;;
    esac

```

**Review by electrocucaracha — COMMENTED** [review]
Hi @vjayaramrh, I besides the suggestions that I let can you include a change into the [platforms section of the molecule file](https://github.com/nephio-project/test-infra/blob/main/e2e/provision/playbooks/roles/bootstrap/molecule/default/molecule.yml#L24-L31)?. I suggest using a [generic](https://app.vagrantup.com/generic) Vagrant box, given they have support for VirtualBox and Libvirt providers.

**Comment by electrocucaracha** [comment]
> Hi @vjayaramrh, I besides the suggestions that I let can you include a change into the [platforms section of the molecule file](https://github.com/nephio-project/test-infra/blob/main/e2e/provision/playbooks/roles/bootstrap/molecule/default/molecule.yml#L24-L31)?. I suggest using a [generic](https://app.vagrantup.com/generic) Vagrant box, given they have support for VirtualBox and Libvirt providers.

I almost forgot, can you also update the [workflow diagram](https://github.com/nephio-project/test-infra/tree/main/e2e/provision/playbooks/roles/bootstrap#workflow)?

File: e2e/provision/init.sh [hunk 2]
```diff
@@ -34,8 +34,15 @@ REPO_DIR=${NEPHIO_REPO_DIR:-$HOME/test-infra}
...
-    apt-get install -y git
+    if [ -f /etc/debian_version ]; then
+            apt-get update
+            apt-get install -y git
+    elif [ -f /etc/redhat-release ]; then
+            dnf install git -y
+    else
+            echo "OS not supported"
+            exit
+    fi
```
[the remark this answers is not in the document]
Comment (radoslawc) [reply e2e/provision/init.sh:45]: 👍

File: e2e/provision/install_sandbox.sh (hunk 1 above)
Comment (radoslawc) [reply e2e/provision/install_sandbox.sh:42]: 👍

**Comment by radoslawc** [comment]
I like Victor's suggestions for OS detection otherwise /lgtm

File: e2e/provision/init.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (vjayaramrh) [reply e2e/provision/init.sh:45]: @electrocucaracha trying to understand what's lacking in the existing check? I find it more readable. What does source do and how is $ID getting populated, it seems cryptic to me.

[the remark this answers is not in the document]
Comment (vjayaramrh) [reply e2e/provision/init.sh:45]: I think I get it now

[the remark this answers is not in the document]
Comment (electrocucaracha) [reply e2e/provision/init.sh:45]: Using a switch/case can make it more readable and enable future additions like openSUSE

File: e2e/provision/playbooks/roles/bootstrap/tasks/load-gtp5g-module.yml [hunk 3]
```diff
@@ -17,20 +17,32 @@
...
+  become: true
+  ansible.builtin.package:
+    name:
+      - kernel-devel
+      - kernel-headers
+      - elfutils-libelf
+      - elfutils-libelf-devel
+    state: present
+  when: ansible_os_family == 'RedHat'
+
```
Comment (electrocucaracha) [inline e2e/provision/playbooks/roles/bootstrap/tasks/load-gtp5g-module.yml:45]:
```suggestion
```

**Review by electrocucaracha — CHANGES_REQUESTED** [review]
@vjayaramrh could you include a platform to validate it with `tox -e bootstrap`?

File: e2e/provision/playbooks/roles/bootstrap/tasks/load-gtp5g-module.yml [hunk 4]
```diff
@@ -17,20 +17,32 @@
...
       skip: true
 
-- name: Install compilation packages
+- name: Install compilation packages on Debian
   become: true
   ansible.builtin.package:
     name: "{{ item }}"
     state: present
   with_items: "{{ build_pkgs }}"
+  when: ansible_os_family == 'Debian'
```
[the remark this answers is not in the document]
Comment (vjayaramrh) [reply e2e/provision/playbooks/roles/bootstrap/tasks/load-gtp5g-module.yml:26]: Great suggestion

File: e2e/provision/playbooks/roles/bootstrap/vars/redhat.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+##############################################################################
+# Copyright (c) 2023 The Nephio Authors.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Apache License, Version 2.0
+# which accompanies this distribution, and is available at
+# http://www.apache.org/licenses/LICENSE-2.0
+##############################################################################
+
+build_pkgs:
+  - make
```
[the remark this answers is not in the document]
Comment (vjayaramrh) [reply e2e/provision/playbooks/roles/bootstrap/vars/redhat.yml:12]: Suggestion implemented

**Comment by vjayaramrh** [comment]
> Hi @vjayaramrh, I besides the suggestions that I let can you include a change into the [platforms section of the molecule file](https://github.com/nephio-project/test-infra/blob/main/e2e/provision/playbooks/roles/bootstrap/molecule/default/molecule.yml#L24-L31)?. I suggest using a [generic](https://app.vagrantup.com/generic) Vagrant box, given they have support for VirtualBox and Libvirt providers.

FYI Suggestion related to VirtualBox and Libvirt providers for Fedora34 implemented

**Comment by vjayaramrh** [comment]
@electrocucaracha Have addressed your comments and suggestions. Would appreciate your review and lgtm
cc: @radoslawc

File: e2e/provision/playbooks/roles/bootstrap/README.md [hunk 6]
```diff
@@ -48,9 +48,9 @@ flowchart TD
     C --> D(Checking host has minimal kernel version)
     D -->|Load gtp5g kernel module| E(Load OS specific variables)
     E --> F(Install compilation packages)
```
Comment (electrocucaracha) [inline e2e/provision/playbooks/roles/bootstrap/README.md:50]:
```suggestion
    E --> F(nstall compilation packages and kernel development tools on Debian/RedHat)
```

**Review by electrocucaracha — COMMENTED** [review]
@vjayaramrh I was running `tox -a bootstrap` and it seems like the [`Update Apt cache`](https://github.com/nephio-project/test-infra/blob/main/e2e/provision/playbooks/roles/bootstrap/molecule/default/prepare.yml#L14-L17) task needs to be changed to include an OS conditional.

**Comment by vjayaramrh** [comment]
> @vjayaramrh I was running `tox -a bootstrap` and it seems like the [`Update Apt cache`](https://github.com/nephio-project/test-infra/blob/main/e2e/provision/playbooks/roles/bootstrap/molecule/default/prepare.yml#L14-L17) task needs to be changed to include an OS conditional.

Done

**Pull request merged**